irohad/model/execution/impl/command_executor.cpp

```diff
@@ -261,6 +262,67 @@ namespace iroha {
       return true;
     }
 
+      // ----------------------------|SubtractAssetQuantity|-----------------------------
+
+      SubtractAssetQuantityExecutor::SubtractAssetQuantityExecutor() {
+        log_ = logger::log("SubtractAssetQuantityExecutor");
+      }
+
+      bool SubtractAssetQuantityExecutor::execute(const Command &command,
+                                             WsvQuery &queries,
+                                             WsvCommand &commands) {
+        auto subtract_asset_quantity = static_cast<const SubtractAssetQuantity &>(command);
+
+        auto asset = queries.getAsset(subtract_asset_quantity.asset_id);
+        if (not asset) {
+          log_->info("asset {} is absent", subtract_asset_quantity.asset_id);
+          return false;
+        }
+        auto precision = asset.value().precision;
+
+        if (subtract_asset_quantity.amount.getPrecision() != precision) {
+          log_->info("amount is wrongly formed");
+          return false;
+        }
+        if (not queries.getAccount(subtract_asset_quantity.account_id).has_value()) {
```

Comment:
This query is not needed, you can remove it. If there is AccountAsset, it means the account exists